testament/testament.nim

```diff
@@ -794,14 +796,16 @@ proc main() =
     p.next
     processPattern(r, pattern, p.cmdLineRest.string, simulate)
   of "r", "run":
+    # "/pathto/tests/stdlib/nre/captures.nim" -> "stdlib" + "tests/stdlib/nre/captures.nim"
     var subPath = p.key.string
-    if subPath.isAbsolute: subPath = subPath.relativePath(getCurrentDir())
+    let nimRoot = currentSourcePath / "../.."
+      # makes sure points to this regardless of cwd or which nim is used to compile this.
+    doAssert existsDir(nimRoot/testsDir) # sanity check
+    if subPath.isAbsolute: subPath = subPath.relativePath(nimRoot)
     # at least one directory is required in the path, to use as a category name
-    let pathParts = split(subPath, {DirSep, AltSep})
-    # "stdlib/nre/captures.nim" -> "stdlib" + "nre/captures.nim"
```

Comment:
please reserve comments

Reply:
I didn't delete this comment, I moved it up (right below `of "r", "run":`) and updated it to reflect the fact it can now be absolute and the fact that the 2nd part produced is tests/stdlib/nre/captures.nim instead of nre/captures.nim